### PR TITLE
Add script bundle for improved script support

### PR DESCRIPTION
**Pull request by danke-sra**
A script bundle contains multiple scripts, each with a target to attach to.
A target can be a scene object, or a special target like GVRScript.

This provides basic support for scene automation, and more sophisticated event/scripting
scenario in the future.

### CODE REVIEW

**Comment by danke-sra** [comment]
The demo app is here: https://github.com/gearvrf/GearVRf-Demos/pull/41

PS. Some commits belong to a pending patch. This is some weekend contribution. Given that scripts and input events are quite usable now, it would be nicer to support more interactivity with scripts.

**Comment by danke-sra** [comment]
Replaced one commit with Rahul's version. Test patch is here: https://github.com/gearvrf/GearVRf-Demos/pull/41

File: GVRf/Framework/src/org/gearvrf/GVRActivity.java [hunk 1]
```diff
@@ -56,6 +56,9 @@
     public static final int KEY_EVENT_MAX = 6;
 
     private GVRViewManager mViewManager;
+    private GVRViewManager mGVRViewManager = null;
```
Comment (liaxim) [inline GVRf/Framework/src/org/gearvrf/GVRActivity.java]: mGVRViewManager and mCamera are unused.

File: GVRf/Framework/src/org/gearvrf/GVRResourceVolume.java [hunk 2]
```diff
@@ -40,11 +68,11 @@
     protected String defaultPath;
     protected boolean enableUrlLocalCache = false;
 
-    /*package*/ GVRResourceVolume(GVRContext gvrContext, VolumeType volume) {
+    public GVRResourceVolume(GVRContext gvrContext, VolumeType volume) {
```
Comment (liaxim) [inline GVRf/Framework/src/org/gearvrf/GVRResourceVolume.java]: The two public ctors could use a javadoc.

**Comment by liaxim** [comment]
Code looks good to me but the sample's performance is not that good. The cursor lags a lot. Also I keep seeing this on exiting the sample app:

```
FATAL EXCEPTION: main
Process: org.gearvrf.sample.gvrjavascript, PID: 8891
java.lang.RuntimeException: MotionEvent { action=ACTION_DOWN, id[0]=0, x[0]=2541.875, y[0]=628.59375, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=13317238, downTime=13317238, deviceId=5, source=0x1002 } recycled twice!
   at android.view.InputEvent.recycle(InputEvent.java:120)
   at android.view.MotionEvent.recycle(MotionEvent.java:1767)
   at android.view.InputEvent.recycleIfNeededAfterDispatch(InputEvent.java:138)
   at android.view.InputEventReceiver.finishInputEvent(InputEventReceiver.java:156)
   at android.view.ViewRootImpl.finishInputEvent(ViewRootImpl.java:6956)
   at android.view.ViewRootImpl.access$1100(ViewRootImpl.java:174)
   at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4388)
   at android.view.ViewRootImpl$SyntheticInputStage.onDeliverToNext(ViewRootImpl.java:5149)
   at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4352)
   at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4329)
```

@rahul27 Any ideas about this "recycled twice" exception?

Looks like disabling the animations in the script brings the performance up the most.

**Comment by rahul27** [comment]
Let me take a look at the sample.

**Comment by rahul27** [comment]
@liaxim I do not see the exception when I try the sample. 

Does it happen very time you exit the sample app? Could you send me the entire log?

Thanks.

**Comment by liaxim** [comment]
Ok, then it could be me. I will retry.

**Comment by danke-sra** [comment]
Rebased. The scene has ~18,000 triangles. For non-animated scene, it shouldn't be slow. But it seems the animation does take cycles off the GL thread. This is due to current GVRAnimationEngine. Nothing in the skeletal animation prevents it from running off the GL thread. We can do some optimization in animation aspects later.

**Comment by thomasflynn** [comment]
Ok, thanks.  This was the information i was looking for.

File: GVRf/Framework/src/org/gearvrf/GVREventReceiver.java [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package org.gearvrf;
```
Comment (thomasflynn) [inline GVRf/Framework/src/org/gearvrf/GVREventReceiver.java]: does this file not have a copyright header?

**Comment by thomasflynn** [comment]
ok, other than the copyright header, any objections to merging this and the associated sample?

**Comment by liaxim** [comment]
Looks good to me.

**Comment by danke-sra** [comment]
Rebased and added copyright header.

**Pull request merged**